src/backend/metalll/src/conversions.rs

```diff
@@ -120,3 +120,62 @@ pub fn map_vertex_format(format: Format) -> Option<MTLVertexFormat> {
         _ => return None,
     })
 }
+
+pub fn map_heap_properties_to_options(properties: memory::HeapProperties) -> MTLResourceOptions {
+    let mut options = MTLResourceOptions::empty();
+    if properties.contains(memory::CPU_VISIBLE) {
+        if properties.contains(memory::COHERENT) {
+            options |= MTLResourceStorageModeShared;
+        } else {
+            options |= MTLResourceStorageModeManaged;
+        }
+    } else if properties.contains(memory::DEVICE_LOCAL) {
+        options |= MTLResourceStorageModePrivate;
+    } else {
+        panic!("invalid heap properties");
+    }
+    if properties.contains(memory::WRITE_COMBINED) {
+        options |= MTLResourceOptionCPUCacheModeWriteCombined;
+    }
+    options
+}
+
+pub fn map_heap_properties_to_storage_and_cache(properties: memory::HeapProperties) -> (MTLStorageMode, MTLCPUCacheMode) {
+    let storage;
+    let cpu;
+    if properties.contains(memory::CPU_VISIBLE) {
```

Comment:
any reason you don't do `let cpu = if {..} else {..};` and the same for `storage`?